dvc/stage/__init__.py

```diff
@@ -339,7 +339,7 @@ def changed_outs(self, allow_missing: bool = False) -> bool:
         for out in self.outs:
             status = out.status()
             if status:
-                if allow_missing and status[str(out)] == "not in cache":
+                if allow_missing and status[str(out)] in ["not in cache", "deleted"]:
```

Comment:
This skips otherwise unchanged stages if the data is available in the local cache